### PR TITLE
[Feature] Provide a way to use Argon instead of Bcrypt for hashing

### PR DESCRIPTION
**Pull request by RomainLanz**
Hey 👋 

This PR completely change how Hash works in Adonis without creating any breaking change.

The new driver system will use `bcrypt` by default if there isn't any configuration (so current codebase will not break).

The new code will let you configure the driver (`bcrypt` or `argon`) via the file `config/hashing.js`. See the example bellow.

```js
// config/hashing.js
module.exports = {
  driver: 'bcrypt',

  bcrypt: {
    round: 10
  }

  argon: {
    timeCost: 4,
    memoryCost: 13,
    parallelism: 2
  }
}
```

If this modification is accepted, I'll provide another PR to add this configuration file to `fullstack`, `api` and `slim` blueprint. I'll also update the current doc to reflect the change.

### CODE REVIEW

**Comment by Atinux** [comment]
Looking forward to it, why it could not be merged already?

File: src/Hash/Facade.js [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+   * @return {Hash}
+   */
+  driver (name) {
+    name = name || this.Config.get('hashing.driver') || 'bcrypt'
+
+    /**
+     * Throw exception when hash.driver is not defined
+     */
+    if (!name) {
+      throw GE.RuntimeException.missingConfig('driver', 'config/hashing.js')
```
Comment (thetutlage) [inline src/Hash/Facade.js]: Can we name the config file as `config/hash.js`, Since we fetch the value as `Config.get(hash.${name})`

**Comment by thetutlage** [comment]
Looks all good to me and just need one change. Since I know @RomainLanz is on holidays, I will pull and add that change myself.

**Comment by thetutlage** [comment]
Released with documentation https://adonisjs.com/docs/4.1/encryption-and-hashing#_hashing_values

**Pull request merged**